### ISSUE TITLE
Node.js deprecation warning in github action

### ISSUE THREAD

**Issue by kalwalt** [facebook/igl#71]
while testing a new build script for Emscripten see issue #67 i noticed this warning:
```
[cmake-build (Ubuntu - Emscripten, ubuntu-latest, Debug, emcc, em++, Ninja)](https://github.com/kalwalt/igl/actions/runs/7783677004/job/21222620454)
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, actions/cache@v2, SimenB/github-actions-cpu-cores@v1. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```
I think actions/checkout need to be upgraded infact the latest is v4 and the igl scripts use v3.

**Comment by corporateshark** [comment]
@kalwalt I noticed there're some recent changes in https://github.com/humbletim/setup-vulkan-sdk/tree/fix-github-actions-deprecations. Once it is merged, please go ahead and create a PR here.

**Comment by kalwalt** [comment]
> @kalwalt I noticed there're some recent changes in https://github.com/humbletim/setup-vulkan-sdk/tree/fix-github-actions-deprecations. Once it is merged, please go ahead and create a PR here.

Yes of course! 🙂

**Comment by kalwalt** [comment]
@corporateshark the PR about setup-vulkan-sdk was merged, so i will send a PR with the updates when i have a bit of time.

**Comment by corporateshark** [comment]
@kalwalt Thank you so much!

**Comment by corporateshark** [comment]
Merged in b4798a5c4d709aa971fe5e361ac64b549a871882.
